drivers/net/ksz9477.c

```diff
@@ -211,6 +219,391 @@ static int ksz9477_sgmii_write_indirect(uint32_t address, uint16_t *value,
   return ret;
 }
 
+static int ksz9477_mmd_read_indirect(ksz9477_port_t port, uint16_t dev,
+                                     uint16_t reg, uint16_t *value,
+                                     uint16_t len)
+{
+  int ret;
+
+  /* Set up MMD device address */
+
+  ret = ksz9477_reg_write16(KSZ9477_PHY_MMD_SETUP(port),
+                            KSZ9477_MMD_OP_MODE_REGISTER | dev);
+  if (ret != OK)
+    {
+      goto out;
+    }
+
+  /* Select register */
+
+  ret = ksz9477_reg_write16(KSZ9477_PHY_MMD_DATA(port), reg);
+  if (ret != OK)
+    {
+      goto out;
+    }
+
+  /* Set MMD operation mode */
+
+  if (len <= 1)
+    {
+      /* no post increment */
+
+      ret = ksz9477_reg_write16(KSZ9477_PHY_MMD_SETUP(port),
+                                KSZ9477_MMD_OP_MODE_NO_INCREMENT | dev);
+      if (ret != OK)
+        {
+          goto out;
+        }
+    }
+  else
+    {
+      /* post increment on reads and writes */
+
+      ret = ksz9477_reg_write16(KSZ9477_PHY_MMD_SETUP(port),
+                                KSZ9477_MMD_OP_MODE_RW_INCREMENT | dev);
+      if (ret != OK)
+        {
+          goto out;
+        }
+    }
+
+  /* Read value */
+
+  while (len-- && ret == OK)
+    {
+      ret = ksz9477_reg_read16(KSZ9477_PHY_MMD_DATA(port), value);
+      value++;
+    }
+
+out:
+
+  return ret;
+}
+
+static int ksz9477_mmd_write_indirect(ksz9477_port_t port, uint8_t dev,
+                                      uint16_t reg, uint16_t *value,
+                                      uint16_t len)
+{
+  int ret;
+
+  /* Set up MMD device address */
+
+  ret = ksz9477_reg_write16(KSZ9477_PHY_MMD_SETUP(port),
+                            KSZ9477_MMD_OP_MODE_REGISTER | dev);
+  if (ret != OK)
+    {
+      goto out;
+    }
+
+  /* Select register */
+
+  ret = ksz9477_reg_write16(KSZ9477_PHY_MMD_DATA(port), reg);
+  if (ret != OK)
+    {
+      goto out;
+    }
+
+  /* Set MMD operation mode */
+
+  if (len <= 1)
+    {
+      /* no post increment */
+
+      ret = ksz9477_reg_write16(KSZ9477_PHY_MMD_SETUP(port),
+                                KSZ9477_MMD_OP_MODE_NO_INCREMENT | dev);
+      if (ret != OK)
+        {
+          goto out;
+        }
+    }
+  else
+    {
+      /* post increment on reads and writes */
+
+      ret = ksz9477_reg_write16(KSZ9477_PHY_MMD_SETUP(port),
+                                KSZ9477_MMD_OP_MODE_RW_INCREMENT | dev);
+      if (ret != OK)
+        {
+          goto out;
+        }
+    }
+
+  /* Write value */
+
+  while (len-- && ret == OK)
+    {
+      ret = ksz9477_reg_write16(KSZ9477_PHY_MMD_DATA(port), *value);
+      value++;
+    }
+
+out:
+
+  return ret;
+}
+
+#if 1
```

Comment:
Maybe leave only what is actially used and remove extra "#if 1" / "#if 0" or at least comment why the dead code is left in?

Reply:
I was pondering that these "official" erratas could be upstreamed..

Reply:
We decided not to upstream these. Drop this. Backup in backup_ksz9477_errata if needed at some point..